### ISSUE TITLE
2mm blade may not be handled properly

### ISSUE THREAD

**Issue by lexelby** [lexelby/inkscape-silhouette#1]
I think the extra-thick (2mm height) blade may not be handled properly by this extension.

The help for the "tool" setting says this:

> 'pen' executes the strokes exactly as sent, 'cut' adds small serifs to help the knive find its orientation at corners.

Try using a pen when selecting "cut" and you'll see what's meant here: the silhouette automatically swings wide around corners.  The blade is offset from center so that it can swivel as it's dragged in different directions.  Swinging wide causes the point of the blade to stay constant as it reorients at a sharp corner.

I believe the radius of the swing is handled <a href="https://github.com/lexelby/inkscape-silhouette/blob/b24f0199f3ecd89d20947d288e5218c12bd7ea6d/silhouette/Graphtec.py#L578">here</a>:

```
cutter = 18
if pen: cutter = 0
s.write("FC%d\x03" % cutter)
```

The value of 18 may only be valid for the standard blade.

**TO DO**: run a usb sniffer while using the extra thick blade with silhouette studio to see if a different FC value is needed.

**Comment by lexelby** [comment]
I can't test this myself because the deep cut blade is not officially supported on the Portrait.  Silhouette Studio won't let me select it as an option in the UI.

**Comment by myselfhimself** [comment]
Hello, I do have a Silhouette Cameo 3, please tell me in details how I can help you.

**Comment by EtherGraf** [comment]
Yes, I stronlgy expect this the case too.
My cutter knifes has an diameter of 0,9mm, wich would give multiplied by 20 the value 18.

**Comment by lexelby** [comment]
Hi @myselfhimself!  What we need is a capture of the USB traffic when you use your Cameo 3.  Instructions for how to do USB sniffing in Windows are [here](https://blog.philippklaus.de/2015/12/silhouette-cameo-2). If you're not in Windows, then I'm not sure how to do it.

Once you have sniffing set up, see if you can record two cuts: one with a normal blade, and one with the "deep cut" blade selected in Silhouette Studio.  Just cut a single straight line, and it doesn't matter if you actually put the deep cut blade in your machine or not.

Post those here and I should be able to analyze them and figure out whether my hypothesis is correct.

Thanks!

**Comment by lexelby** [comment]
@EtherGraf Thanks, that's really helpful.  I kind of guessed my blade was 0.9mm, but it's pretty hard to measure just by eye.  Did you use callipers?

**Comment by EtherGraf** [comment]
Yes, today checked by calliper, but I had known it before. I use not the original blades, but fully compatible replacement ones, Type CB09 with a diameter of 0.9mm.

**Comment by EtherGraf** [comment]
Just played with this value and a pen inside the plotter. 
As assumed this is the circle diameter, which is taken to adjust the blade to the new direction on corners. 
It seems to be limited to the value 50, values above shows an identical result as value 50. Values between 0 and 50 increases the diamter proportional.

**Comment by EtherGraf** [comment]
I just have added a option for blade diameter in the fablabnbg repository.

**Comment by lexelby** [comment]
Nice, thanks! Now we just need to know the diameter of the deep cut blade.

As a note, this will mean that inkscape-silhouette can do something that Silhouette Studio can't: allow you to use the deep cut blade with a Portrait machine.  Pretty nifty :)
